feature_reviser/transformer/custom_transformer.py

```diff
@@ -232,25 +259,79 @@ def __to_float(ip4_devisor: float, ip6_devisor: float, ip_address: str) -> float
             except:  # pylint: disable=W0702
                 return -1
 
-    def fit(self) -> "IPAddressEncoderTransformer":
+
+class EmailTransformer(BaseEstimator, TransformerMixin):
+    """
+    Transforms an email address into multiple features.
+
+    Args:
+        new_column_name (str): The name of the output column.
+    """
+
+    def fit(self, X=None, y=None) -> "EmailTransformer":  # type: ignore
         """
         No need to fit anything.
         """
         return self
 
     def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         """
-        Transforms the column containing the IP addresses to float column.
-        `-1` indicates that the value could not be parsed.
+        Transforms the one column from X, containing the email addresses, into multiple columns.
 
         Args:
             X (pandas.DataFrame): DataFrame to transform.
 
         Returns:
-            pandas.DataFrame: Transformed dataframe.
+            pandas.DataFrame: Transformed dataframe containing the extra columns.
         """
-        function = functools.partial(
-            IPAddressEncoderTransformer.__to_float, self.ip4_devisor, self.ip6_devisor
+
+        X = X.copy()
+
+        if X.shape[1] != 1:
+            raise ValueError(
+                "Only one column is allowed! Please try something like: `df[['email']]`."
+            )
+        column_name = X.iloc[:, 0].name
+
+        X[f"{column_name}_domain"] = (
+            X.iloc[:, 0].str.split("@").str[1].str.split(".").str[1]
```

Comment:
```suggestion
            X.iloc[:, 0].str.split("@").str[1].str.split(".").str[0]
```
The previous line returns the top-level domain, e.g. ".com" of an email address. I suppose we want to extract only the first part of the domain, e.g. "gmail".